### PR TITLE
Replace `loadable-components` with `React.lazy`.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is a minor optimization to remove the need for an external dependency like `loadable-components` and using stock React functionality instead.

## How Has This Been Tested?

The web app was build and tested in the browser while looking at the network tab to ensure that code splitting is still working.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/routing/loadAsync.tsx [hunk 1]
```diff
@@ -14,22 +14,60 @@
...
 
 type Props = {
-  error: {
-    message: string;
-  };
+  error: Error;
 };
 
-const ErrorComponent: React.FC<Props> = ({ error }: Props) => <div>Loading component failed: {error.message}</div>;
+const ErrorComponent: React.ComponentType<Props> = ({ error }: Props) => <div>Loading component failed: {error.message}</div>;
```
Comment (Gaya) [inline graylog2-web-interface/src/routing/loadAsync.tsx:31]: Isn't this a `React.FunctionComponent`?  Why did you change it into a `React.ComponentType`?

**Review by Gaya — APPROVED** [review]
Great! Much better this way. Just a small question and one test is failing on my end (and jenkins)

File: graylog2-web-interface/src/routing/loadAsync.tsx (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/routing/loadAsync.tsx:31]: `React.ComponentType` is the abstraction, it covers both class and functional components. We do not really care in this case if it's a class or functional component, so I captured that in its type.

Comment (Gaya) [reply graylog2-web-interface/src/routing/loadAsync.tsx:31]: Makes sense! Thanks for clarifying, never used / seen it before.

**Review by Gaya — APPROVED** [review]
LGTM!
